hudi-common/src/main/java/org/apache/hudi/metadata/HoodieTableMetadataUtil.java

```diff
@@ -330,78 +322,67 @@ public static void deleteMetadataTable(String basePath, HoodieEngineContext cont
       });
     });
 
-    return engineContext.map(deleteFileList, deleteFileInfo -> {
-      return HoodieMetadataPayload.createBloomFilterMetadataRecord(
-          deleteFileInfo.getLeft(), deleteFileInfo.getRight(), instantTime, ByteBuffer.allocate(0), true);
-    }, 1).stream().collect(Collectors.toList());
+    HoodieData<Pair<String, String>> deleteFileListRDD = engineContext.parallelize(deleteFileList,
```

Comment:
Do we really need to cast this action t/h RDD? Do we envision that this will scale past the point when we won't be able to handle this on the driver? 

I'm worried about serialization cost we incur for every record we handle t/h RDD (serializing/de closure) to be able to create a single object